SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -1,271 +1,215 @@
+			break;
+		case SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID:
+			Components.Intent = AuthorityIntent(ComponentData.GetUnderlying());
+			break;
+		case SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID:
+			Components.Presence = ComponentPresence(ComponentData.GetUnderlying());
+			break;
+		case SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID:
+			Components.OwningClientWorker = NetOwningClientWorker(ComponentData.GetUnderlying());
+			break;
+		default:
+			break;
+		}
+	}
 }
 
-bool SpatialLoadBalanceEnforcer::CanEnforce(Worker_EntityId EntityId) const
+bool SpatialLoadBalanceEnforcer::ApplyComponentUpdate(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId,
+													  Schema_ComponentUpdate* Update)
 {
-	// We need to be able to see the ACL component
-	return StaticComponentView->HasComponent(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID)
-		   // and the authority intent component
-		   && StaticComponentView->HasComponent(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
-		   // and the component presence component
-		   && StaticComponentView->HasComponent(EntityId, SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID)
-		   // and we have to be able to write to the ACL component.
-		   && StaticComponentView->HasAuthority(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+	switch (ComponentId)
+	{
+	case SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID:
+		DataStore[EntityId].Intent.ApplyComponentUpdate(Update);
+		return true;
+	case SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID:
+		DataStore[EntityId].Presence.ApplyComponentUpdate(Update);
+		return true;
+	case SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID:
+		DataStore[EntityId].OwningClientWorker.ApplyComponentUpdate(Update);
+		return true;
+	default:
+		break;
+	}
+	return false;
 }
 
-bool SpatialLoadBalanceEnforcer::HandlesComponent(Worker_ComponentId ComponentId) const
+bool SpatialLoadBalanceEnforcer::ApplyComponentRefresh(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId,
+													   Schema_ComponentData* Data)
 {
-	return ComponentId == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || ComponentId == SpatialConstants::ENTITY_ACL_COMPONENT_ID
-		   || ComponentId == SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID
-		   || ComponentId == SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID;
+	switch (ComponentId)
+	{
+	case SpatialConstants::ENTITY_ACL_COMPONENT_ID:
+		DataStore[EntityId].Acl = EntityAcl(Data);
+		break;
+	case SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID:
+		DataStore[EntityId].Intent = AuthorityIntent(Data);
+		return true;
+	case SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID:
+		DataStore[EntityId].Presence = ComponentPresence(Data);
+		return true;
+	case SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID:
+		DataStore[EntityId].OwningClientWorker = NetOwningClientWorker(Data);
+		return true;
+	default:
```

Comment:
Perhaps we want some checks here if default is invalid?

Reply:
the subview gives us the whole entity if it's complete, so this default could be all the other components other than the LB components. We either default here, or enforce that we only enter this function with correct components

Reply:
Yup super!